general/mem_manager.hpp

```diff
@@ -57,12 +59,14 @@ extern const char *MemoryTypeName[MemoryTypeSize];
  *  use MemoryClass::DEVICE for their inputs. */
 enum class MemoryClass
 {
-   HOST,    /**< Memory types: { HOST, HOST_32, HOST_64, HOST_DEBUG,
-                                 HOST_UMPIRE, MANAGED } */
-   HOST_32, ///< Memory types: { HOST_32, HOST_64, HOST_DEBUG }
-   HOST_64, ///< Memory types: { HOST_64, HOST_DEBUG }
-   DEVICE,  ///< Memory types: { DEVICE, DEVICE_DEBUG, DEVICE_UMPIRE, MANAGED }
-   MANAGED  ///< Memory types: { MANAGED }
+   HOST,        /**< Memory types: { HOST, HOST_32, HOST_64, HOST_DEBUG,
+                                     HOST_UMPIRE, MANAGED } */
+   HOST_32,     ///< Memory types: { HOST_32, HOST_64, HOST_DEBUG }
+   HOST_64,     ///< Memory types: { HOST_64, HOST_DEBUG }
+   DEVICE,      /**< Memory types: { DEVICE, DEVICE_DEBUG, DEVICE_UMPIRE,
+                                     MANAGED } */
+   DEVICE_TEMP, ///< Memory types: { DEVICE_UMPIRE }
```

Comment:
Why is there a correspondence between DEVICE_TEMP and DEVICE_UMPIRE ?

Reply:
Because the `DEVICE_UMPIRE` type implements a temporary memory allocator (`AllocTemp`)